### PR TITLE
[4] Cleaner Loading Component

### PR DESCRIPTION
**Pull request by aivantg**
[//]: # "These comments meant for your reference, they are invisible and don't need to be deleted"

## What's new in this PR?

[//]: # '#####################################'
[//]: # '####### YOUR ANSWER BELOW ###########'

- Created a new wrapper around Route to only display route if app is not loading any data! 
- Cleaned up all components to not think about app data loading!

[//]: # '############## END ##################'

### How to Review

[//]: # 'The order in which to review files and 
what to expect when testing locally'
[//]: # '#####################################'
[//]: # '####### YOUR ANSWER BELOW ###########'

Try to explore all the screens to make sure they load properly!

[//]: # '############## END ##################'

### Related PRs

[//]: # "Optional - related PRs you're waiting on
/ PRs that will conflict, etc"
[//]: # '#####################################'
[//]: # '####### YOUR ANSWER BELOW ###########'

Branches off of #124 

[//]: # '############## END ##################'

### Next steps

[//]: # "What doesn't work yet, what's NOT in this 
PR/has to be done "
[//]: # '#####################################'
[//]: # '####### YOUR ANSWER BELOW ###########'
[//]: # '############## END ##################'

### Tests Performed, Edge Cases

[//]: # '#####################################'
[//]: # '####### YOUR ANSWER BELOW ###########'
[//]: # '############## END ##################'

### Screenshots

[//]: # '#####################################'
[//]: # '#### YOUR SCREENSHOTS BELOW ########'
[//]: # '############## END ##################'

CC: @dfangshuo

### CODE REVIEW

File: src/components/PPRoute.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
+/* eslint-disable react/jsx-props-no-spreading */
+import React from 'react';
+import { Route } from 'react-router-dom';
+import { connect } from 'react-redux';
+import LoadingComponent from './LoadingComponent';
+
+const PPRoute = ({ isLoadingUserData, isLoadingAnnouncements, ...rest }) => {
+  const loading = isLoadingAnnouncements || isLoadingUserData;
+  return loading ? <LoadingComponent /> : <Route {...rest} />;
```
Comment (dfangshuo) [inline src/components/PPRoute.js:9]: nit: i found it easier to read if you just did 
```suggestion
  return isLoadingAnnouncements || isLoadingUserData; ? <LoadingComponent /> : <Route {...rest} />;
```
